components/FA2Transfer.tsx

```diff
@@ -110,13 +123,15 @@ const FA2Transfer = ({
         .then((v: fa2Token[]) => {
           setCanSeeMore(v.length === FETCH_COUNT);
 
-          return Promise.resolve(v);
+          return Promise.resolve(
+            v.filter(token => !toExclude.includes(token.id))
+          );
         }),
     [state.currentContract]
   );
 
   useEffect(() => {
-    const value = getFieldProps(makeName("token")).value;
+    const value = getFieldProps(makeName("token")).value as fa2Token;
```

Comment:
Basically it can't happen as `formValue[]` is valid for this name: `transfers.${proposalIndex}.values` and not for `transfers.${proposalIndex}.values.${index}.token`. But this is the only way to avoid an `any` here: `FieldInputProps<fa2Token | formValue[] | undefined>`

Reply:
but does it mean it could never be `undefined` as well? we tell compile `value` can only be `fa2Token`. Therefore, we don't need line #139 ` if (!value) return;`

so if in ocaml/haskell, we would do pattern match for each case, like
```
match ... with
| fa2Token -> ...
| formValue[] -> ...
| undefined -> ...
```

to make sure each case is handled properly.  If some cases are not going to happen, throwing error message is a good idea. sometimes, error is not only for users, it's also developers.  as last time, you accidentally remove the input field in contract execution. if there isn't an error throwing, we may end up error report by users.